integrations/pgvector/tests/test_document_store.py

```diff
@@ -237,3 +237,12 @@ def get_table_oid(document_store, schema_name, table_name):
     # Clean up: drop the schema after the test
     with psycopg.connect(connection_string, autocommit=True) as conn:
         conn.execute(f"DROP SCHEMA IF EXISTS {schema_name} CASCADE")
+
+
+@pytest.mark.integration
+def test_delete_table_first_call(document_store):
+    """
+    should be able to call _ensure_db_setup() inside itself while the connection is not yet established, and
+    should not throw any exceptions.
```

Comment:
I would write something similar to the following:
```suggestion
    Test that if `delete_table can be executed as the initial operation on the Document Store 
    without triggering errors due to an uninitialized state.
```